src/Compilers/Core/MSBuildTask/Csc.cs

```diff
@@ -169,7 +169,7 @@ public int WarningLevel
 
         #region Tool Members
 
-        private static readonly string[] s_separators = { Environment.NewLine };
+        private static readonly string[] s_separators = { "\r\n", "\r", "\n" };
```

Comment:
nit: may be worth a comment pointing to MoveLinesFromStringBuilderToMessageQueue

Reply:
Good idea. Though I thought instead of referencing implementation details of .Net, it would be better to document the behavior publicly, so I opened https://github.com/dotnet/dotnet-api-docs/pull/4644.